csharp/src/Microsoft.ML.OnnxRuntime/OrtValue.shared.cs

```diff
@@ -263,12 +263,16 @@ public ReadOnlyMemory<char> GetStringElementAsMemory(int index)
         /// <returns>UTF-16 string instance</returns>
         public string GetStringElement(int index)
         {
-            var chars = GetStringTensorElementChars(index);
-            if (chars.Length == 0)
+            GetStringTensorElementBuffer((UIntPtr)index, out uint bytesLen, out IntPtr bufferPtr);
+            if (bytesLen == 0)
             {
                 return string.Empty;
             }
-            return new string(chars);
+
+            unsafe
+            {
+                return new string((sbyte*)bufferPtr.ToPointer(), 0, (int)bytesLen, Encoding.UTF8);
```

Comment:
looking into the string ctor - does `bytesLen` correspond to the number of characters? with utf8 it may be more, right?
https://learn.microsoft.com/en-us/dotnet/api/system.string.-ctor?view=net-8.0#system-string-ctor(system-sbyte*-system-int32-system-int32-system-text-encoding)

Reply:
The name is correct, it corresponds to the number of `UTF-8` bytes, not characters.

Reply:
Checked with .NET, it is actually bytes, but I replaced the line with an easier to use API